### PR TITLE
add: Configure secret providers for Apache Kafka Connect

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Describe your changes
Added content for configuring AWS Secrets Manager and HashiCorp Vault as secret providers in Apache Kafka Connect on Aiven, including examples for JDBC connectors.

Ref: [EC-343](https://aiven.atlassian.net/browse/EC-343)

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

[EC-343]: https://aiven.atlassian.net/browse/EC-343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/products/kafka/kafka-connect/howto/configure-aws-secrets-manager.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  target database if it does not exist.
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+Reference secrets in the JDBC sink connector configuration using the following
+CLI command:
+
+```bash
+avn service connector create                                    \
```
Comment (keejon) [inline docs/products/kafka/kafka-connect/howto/configure-aws-secrets-manager.md:143]: I think this is not valid cli usage. Have a look at: https://aiven.io/docs/tools/cli/service/connector.

```bash
avn service connector create --help
usage: avn service connector create [-h] [--project PROJECT] service_name connector_config

Create a Kafka connector

positional arguments:
  connector_config   JSON string or path (preceded by '@') to a JSON configuration file
  service_name       Service name

options:
  -h, --help         show this help message and exit
  --project PROJECT  Project name to use, default None
```

File: docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  the target database if it does not exist.
+
+</TabItem>
+<TabItem value="cli" label="CLI">
+
+Reference secrets in the JDBC sink connector configuration using the following
+CLI command:
+
+```bash
+avn service connector create \
```
Comment (keejon) [inline docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md:139]: Same here

File: docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+  --header 'Content-Type: application/json' \
+  --data '{
+    "user_config": {
+      "secret_providers": [
+        {
+          "name": "vault",
+          "vault": {
+            "auth_method": "token",
+            "address": "https://vault.aiven.fi:8200/
+            ",
```
Comment (keejon) [inline docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md:44]: remove newline

File: docs/products/kafka/kafka-connect/howto/configure-aws-secrets-manager.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+After configuring the secret providers, you can reference secrets in your
+connector configurations.
+
+### JDBC sink connector
+
+<Tabs groupId="reference-secrets-sink">
+<TabItem value="api" label="API" default>
+
+Configure a JDBC sink connector using the API with secrets referenced from
```
Comment (MichaelTansiniAiven) [inline docs/products/kafka/kafka-connect/howto/configure-aws-secrets-manager.md:109]: is 'the API' standard for Aiven's Restful API?

File: docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+</TabItem>
+</Tabs>
+
+## Reference secrets in connector configurations
+
+After configuring the secret providers, you can reference secrets in your
+connector configurations.
+
+### JDBC sink connector
```
Comment (MichaelTansiniAiven) [inline docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md:101]: the JDBC example is illustrative right? Any connector we have can support this @keejon ?

**Review by MichaelTansiniAiven — APPROVED** [review]
All looks good, just one question about the examples for Jonas

File: docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md (hunk 5 above)
Comment (keejon) [reply docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md:100]: jup, can be used in any connector 👍

Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/configure-hashicorp-vault.md:101]: Updated the text to clarify this.

File: docs/products/kafka/kafka-connect/howto/configure-aws-secrets-manager.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
+---
+title: Configure AWS Secrets Manager
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Configure and use AWS Secrets Manager as a secret provider in Aiven for Apache Kafka® Connect services.
```
Comment (ArthurFlag) [inline docs/products/kafka/kafka-connect/howto/configure-aws-secrets-manager.md:8]:
```suggestion
Configure and use [AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html) as a secret provider in Aiven for Apache Kafka® Connect services.
```

**Review by ArthurFlag — CHANGES_REQUESTED** [review]
Just a ux nitpick, but the rest looks clean 👌

**Review by ArthurFlag — APPROVED** [review]

**Pull request merged**